Dockerfile

```diff
@@ -1,7 +1,28 @@
-FROM ubuntu:latest
+# CHIA BUILD STEP
+FROM python:3.9 AS chia_build
 
-EXPOSE 8555
-EXPOSE 8444
+RUN dpkg-reconfigure -f noninteractive tzdata
+
+ARG BRANCH=latest
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+        lsb-release sudo
+
+RUN echo "cloning ${BRANCH}" && \
+    git clone --branch ${BRANCH} https://github.com/Chia-Network/chia-blockchain.git 
```

Comment:
The `git clone` here and the `git submodule update` later can be combined into a single command: `git clone --recurse-submodules=mozilla-ca ...`. Saves another layer.

Reply:
The layers in the `chia_build` step are discarded and do not contribute to the final image.

The final `COPY` at [line 45](https://github.com/Chia-Network/chia-docker/pull/130/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R45) will put the resulting files from the build in a single layer at final image assembly.

Reply:
I rechecked with the original line restored with the original lines: https://github.com/Chia-Network/chia-docker/blob/195d313601747b67aa8a14f880c33fd3d21120a7/Dockerfile#L25-L29 .

Resulting final image size was exactly the same.

I can revert the lines back if requested.

Reply:
Yeah, the saved layer won't have any effect on image size. I personally still consider reducing the number of layers worthwhile (esp. if it's easily achievable) and even if it's only build images. (Those of us regularly building images manually typically keep the build images around, for faster cycle times.)

But it's definitely a topic with the lowest of priorities :)

So I'd still suggest using the simplified:
```
RUN echo "cloning ${BRANCH}" && \
    git clone --branch ${BRANCH} --recurse-submodules=mozilla-ca https://github.com/Chia-Network/chia-blockchain.git 
```
But all other variants (two RUNs, as in the PR at the moment, or the original lines) are certainly fine as well.

Reply:
That does look a lot cleaner.

@cmmarslender : Do you want me to add a commit with this change?

Reply:
Yeah thats fine

Reply:
New range of code:
https://github.com/Chia-Network/chia-docker/blob/9221eb7e2fcf17c1b0b2d4423540b02608e3a273/Dockerfile#L10-L15

I managed  to shave off one more layer by moving `WORKDIR` up and having `git` clone inside the working directory.